tests/core/test_save_restore.py

```diff
@@ -85,12 +101,66 @@ def setup_validation_data(self, val_data_config: Union[DictConfig, Dict]):
     def setup_test_data(self, test_data_config: Union[DictConfig, Dict]):
         self._test_dl = None
 
+    @classmethod
     def list_available_models(cls):
         return []
 
 
+class MockModelWithChildren(MockModel):
+    """
+    Mock Model, can contain 2 children (other NeMo models)
+    """
+
+    def __init__(self, cfg, trainer=None):
+        super().__init__(cfg=cfg, trainer=trainer)
+
+        # child 1
+        if cfg.get('child1_model_path') is None and cfg.get('child1_model') is None:
+            self.child1_model = None
+        else:
+            # if `child1_model_path` is set, model should be restored from nemo checkpoint
+            # otherwise model is constructed from config in `child1_model`
+            if cfg.get('child1_model_path') is not None:
+                self.child1_model = ModelPT.restore_from(cfg.child1_model_path)
+                # path in config should be set to `None` to restore model from config after saving
+                self.cfg.child1_model_path = None
+                # config for child model should be stored in base model config
+                self.cfg.child1_model = self.child1_model.cfg
+            else:
+                self.child1_model = ModelPT.from_config_dict(self.cfg.child1_model)
+
+        # child 2
+        if cfg.get('child2_model_path') is None and cfg.get('child2_model') is None:
+            self.child2_model = None
+        else:
+            # if `child2_model_path` is set, model should be restored from nemo checkpoint
```

Comment:
"nemo checkpoint" -> "nemo model" ?

Reply:
This is a checkpoint, so I changed it to ".nemo model checkpoints"